src/gui/qgslayermetadatasearchwidget.h

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************
+  qgslayermetadatasearchwidget.h - QgsLayerMetadataSearchWidget
+
+ ---------------------
+ begin                : 1.9.2022
+ copyright            : (C) 2022 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSLAYERMETADATASEARCHWIDGET_H
+#define QGSLAYERMETADATASEARCHWIDGET_H
+
+#include "qgis_gui.h"
+#include <QWidget>
+#include "ui_qgslayermetadatasearchwidget.h"
+#include "qgsfeedback.h"
+#include "qgsabstractlayermetadataprovider.h"
+
+class QgsMapCanvas;
+class QgsLayerMetadataResultsProxyModel;
+
+/**
+ * \ingroup gui
+ * \brief The QgsLayerMetadataSearchWidget class offers layer metadata search and filtering.
+ * It is designed to be embedded in the data source manager dialog.
```

Comment:
Can't we keep this widget generic enough to be used elsewhere too? Either that or SIP_SKIP it so we can later.